### PR TITLE
Add docs for shell completion

### PR DESCRIPTION
**Pull request by gzdunek**
Initially, the shell completion was meant to work only in tsh, but under the hood we use the same library, so it can work with any CLI tool. Because of that I decided to put the instructions in `cli.mdx`. If you think that there is a better place for it, I'm happy to move it.

### CODE REVIEW

File: docs/pages/includes/cli-completion.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<Details title="Improve the CLI experience: enable shell completion">
+
+  Out of the box, CLI tools can take advantage of completion hints for flags and commands.
+  For example, typing `tsh` and pressing `Tab` will show all available subcommands, typing `tsh --` and pressing `Tab`
+  will show all available flags.
+
+  Supported shells are bash and zsh.
+
+  To enable completion add an additional statement to `bash_profile` (or equivalent).
+  Replace `<cli-tool>` with `teleport`, `tsh`, `tctl` or `tbot`.
```
Comment (gzdunek) [inline docs/pages/includes/cli-completion.mdx:10]: I'm not sure if `<` `>` are needed (do we have a convention for that?).

File: docs/pages/includes/cli-completion.mdx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<Details title="Improve the CLI experience: enable shell completion">
+
+  Out of the box, CLI tools can take advantage of completion hints for flags and commands.
```
Comment (zmb3) [inline docs/pages/includes/cli-completion.mdx:3]:
```suggestion
Teleport's CLI tools can provide completion hints for bash and zsh.
```

And then remove the "supported shells" section below.

File: docs/pages/includes/cli-completion.mdx (hunk 1 above)
Comment (zmb3) [reply docs/pages/includes/cli-completion.mdx:10]: I would just be specific (use `tsh` as the example), and then add a sentence at the very end that you can repeat the same process for `tctl`, `teleport`, and `tsh`.

(In practice, most people will only use this for tsh and tctl, teleport and tbot are typically run as daemons and not often invoked manually.

File: docs/pages/reference/cli.mdx [hunk 3]
```diff
@@ -12,6 +12,8 @@ Teleport is made up of four CLI tools.
 
 (!docs/pages/includes/permission-warning.mdx!)
 
+(!docs/pages/includes/cli-completion.mdx!)
```
Comment (zmb3) [inline docs/pages/reference/cli.mdx:15]: If the partial is only included in one place, should we just inline the content directly here?

**Review by zmb3 — APPROVED** [review]

**Review by alexfornuto — CHANGES_REQUESTED** [review]
What version of `tsh` introduces `--completion-script-bash/zsh`? Using `v13.0.0 git:v13.0.0-0-g6d0ce2bf` I don't have either flag:

```sh
ERROR: unknown long flag '--completion-script-bash'
```

EDIT: Same for `tctl`:

```sh
tctl: error: unknown long flag '--completion-script-bash'
```

**Comment by zmb3** [comment]
> What version of `tsh` introduces `--completion-script-bash/zsh`? Using `v13.0.0 git:v13.0.0-0-g6d0ce2bf` I don't have either flag:
> 
> ```shell
> ERROR: unknown long flag '--completion-script-bash'
> ```
> 
> EDIT: Same for `tctl`:
> 
> ```shell
> tctl: error: unknown long flag '--completion-script-bash'
> ```

It only exists in master right now. See https://github.com/gravitational/teleport/pull/26238

File: docs/pages/reference/cli.mdx [hunk 4]
```diff
@@ -12,6 +12,33 @@ Teleport is made up of four CLI tools.
...
 
+<Details title="Improve the CLI experience: enable shell completion">
+
+  Teleport's CLI tools can provide completion hints for bash and zsh.
+  For example, typing `tsh` and pressing `Tab` will show all available subcommands, typing `tsh --` and pressing `Tab`
+  will show all available flags.
+
+  To enable completion add an additional statement to `bash_profile` (or equivalent).
+
+  Example (`.bashrc`):
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply docs/pages/reference/cli.mdx:23]: What do you think about removing specific names:
"To enable completion, add an additional statement to your shell configuration file." 

And then in the examples there are `.bashrc` and `.zshrc`.

**Comment by alexfornuto** [comment]
> @alexfornuto how can I ignore spellcheck errors for the shell examples?

I would just add `autoload` and `compinit` to the 

> @alexfornuto how can I ignore spellcheck errors for the shell examples?

Added the terms in e0d2881d12c0ae5657cf17849bf1dbc4d0c7ad58

**Review by alexfornuto — APPROVED** [review]

**Comment by alexfornuto** [comment]
> It only exists in master right now. See #26238

FYI The latest build works for me for `tsh`, but not `tctl`:

```
❯ ./build/tctl --shell-completion-bash
tctl: error: unknown long flag '--shell-completion-bash'
❯ ./build/tctl version
Teleport v14.0.0-dev git: go1.20.3
```

**Review by alexfornuto — APPROVED** [review]

**Comment by gzdunek** [comment]
I will create docs backports when I backport the main feature: https://github.com/gravitational/teleport/pull/26238.

**Pull request merged**